components/merkledb/src/views/mod.rs

```diff
@@ -348,6 +348,30 @@ impl<T: RawAccessMut> View<T> {
     }
 }
 
+//TODO: Wrapper to prevent abstraction leak. Maybe there is better way.
+pub(crate) struct IndexesPoolWrapper<T: RawAccess> {
```

Comment:
~Abstractions shmabstractions~ `IndexesPool` is already made crate-private by the migration PR, so this looks redundant.